classes/privacy/provider.php

```diff
@@ -14,31 +14,167 @@
+                        'count' => 'privacy:metadata:local_wunderbyte_table:count',
+                ],
+                'privacy:metadata:local_wunderbyte_table'
+        );
+        return $collection;
+    }
+
+
+    /**
+     * Get the list of contexts that contain user information for the specified user.
+     *
+     * @param int $userid the userid.
+     * @return contextlist the list of contexts containing user info for the user.
+     */
+    public static function get_contexts_for_userid(int $userid): contextlist {
+        $contextlist = new contextlist();
+        $contextlist->add_system_context();
+        return $contextlist;
+    }
+
+    /**
+     * Export personal data for the given approved_contextlist. User and context information is contained within the contextlist.
+     *
+     * @param approved_contextlist $contextlist a list of contexts approved for export.
+     */
+    public static function export_user_data(approved_contextlist $contextlist) {
+        global $DB;
+        $userid = $contextlist->get_user()->id;
+        foreach ($contextlist as $context) {
+            if ($context->contextlevel == CONTEXT_SYSTEM) {
+                $i = 0;
+                $data = [];
+
+                foreach (self::$tables as $table) {
+                    $rows = $DB->get_records($table, ['userid' => $userid]);
+                    if (count($rows) > 0) {
+                        foreach ($rows as $row) {
+                            $data[$i] = $row;
+                            $i++;
+                        }
+                    }
+                }
+
+                writer::with_context($context)->export_data(
+                        [get_string('privacy:metadata:local_wunderbyte_table', 'local_wunderbyte_table')],
+                        (object) $data);
+            }
+        }
+    }
+
+    /**
+     * Delete all data for all users in the specified context.
+     *
+     * @param \context $context the context to delete in.
+     */
+    public static function delete_data_for_all_users_in_context(\context $context) {
+        global $DB;
+        if ($context->contextlevel == CONTEXT_SYSTEM) {
+            foreach (self::$tables as $table) {
+                // Delete the records from table
+                $DB->delete_records($table, null);
+            }
+        }
+    }
+
+    /**
+     * Delete all user data for the specified user, in the specified contexts.
+     *
+     * @param approved_contextlist $contextlist a list of contexts approved for deletion.
+     */
+    public static function delete_data_for_user(approved_contextlist $contextlist) {
+        global $DB;
+        $userid = $contextlist->get_user()->id;
+
+        foreach ($contextlist as $context) {
+            if ($context->contextlevel == CONTEXT_SYSTEM) {
+                foreach (self::$tables as $table) {
+                    $sql = "DELETE FROM {$table} WHERE userid = :userid";
+                    $DB->execute($sql, ['userid' => $userid]);
```

Comment:
Would be more appropriate to use the inbuilt $DB->delete_records('local_wunderbyte_table', ['userid' => $userid]);